src/builder-context.h

```diff
@@ -22,14 +22,22 @@
 #define __BUILDER_CONTEXT_H__
 
 #include <gio/gio.h>
-#include <libsoup/soup.h>
 #include <curl/curl.h>
 #include "builder-options.h"
 #include "builder-utils.h"
 #include "builder-sdk-config.h"
 
 G_BEGIN_DECLS
 
+/* Same as SOUP_HTTP_URI_FLAGS, means all possible flags for http uris */
+
+#if GLIB_CHECK_VERSION (2, 68, 0) || !GLIB_CHECK_VERSION (2, 66, 0)
```

Comment:
I'm not saying we care but its worth noting CURL has a complete API to parse URIs and GUri does nothing unique for us.

But of course the CURL API is the CURL API so its a lot uglier than GUri. https://curl.se/libcurl/c/libcurl-url.html

Reply:
Considering large amounts of this code is almost copy-paste from flatpak porting away from libsoup2, I would prefer not going into a larger refactoring into CURL API which I don't know well.